### PR TITLE
Standardize type conversions

### PR DESCRIPTION
**Pull request by DavidKnott**
### - What I did
Standardize type conversions, implementing VIP #629 

### - How I did it
Broke out type conversion into its own file and combined the different ways to convert between types into one:
```
convert(input, 'output_type')
x: num
y: decimal = convert(x, 'decimal`)
```
### - How to verify it
Look at `test_convert.py`
### - Description for the changelog
Standardize type conversions.
### - Cute Animal Picture
image

### CODE REVIEW

**Comment by fubuloubu** [comment]
Love it! Reviewing...

**Comment by fubuloubu** [comment]
I remember discussing putting the type in quotes, but I just had a conflicting thought (perhaps for later on down the road). When syntax highlighting is developed for vyper, it would actually be more apparent to look at `y: num256 = convert(x, num256)` than to have `num256` in quotes, since there will be syntax highlighting for built-in types

**Comment by DavidKnott** [comment]
Merging this.

File: vyper/functions/functions.py [hunk 1]
```diff
@@ -156,47 +71,62 @@ def floor(expr, args, kwargs, context):
...
+    #             ['mload', ['add', args[0], 32]], typ=BaseType('bytes32')
+    #             )
+    #         elif args[0].location == "storage":
+    #             return LLLnode.from_list(
+    #                 ['sload', ['add', ['sha3_32', args[0]], 1]], typ=BaseType('bytes32')
+    #             )
+    #     else:
+    #         return LLLnode(value=args[0].value, args=args[0].args, typ=BaseType('bytes32'), pos=getpos(expr))
+    # else:
+    #     raise Exception("Conversion of {} to {} is not allowed.".format(typ, args[1]))
```
Comment (jacqueswww) [inline vyper/functions/functions.py]: @DavidKnott I think we should remove this comment block now?

**Pull request merged**